src/formula/callable_objects.cpp

```diff
@@ -173,6 +176,10 @@ int attack_type_callable::do_compare(const formula_callable* callable) const
 		return att_->type().compare(att_callable->att_->type());
 	}
 
+	if(att_->effective_damage_type().first != att_callable->att_->effective_damage_type().first) {
+		return att_->effective_damage_type().first.compare(att_callable->att_->effective_damage_type().first);
+	}
```

Comment:
What is the purpose of this check?

Reply:
@Pentarctagon @CelticMinstrel if checking the type actually used in a formula is problematic, I can remove it, I just wanted to harmonize the language with [filter_weapon]

Reply:
formula modification removed

Reply:
And yet my question remains unanswered.

Reply:
@Pentarctagon This check is the code to check the type of damage actually inflicted while the old code for type() is reassigned to 'original_type', in short if you use 'type' in a formula in [filter_attack/weapon], it is the type of damage that will actually be used that will be detected, a bit like if you use [filter_attack]type=fire which matches with fire type attacks even if the attack is modified by [damage_type]alternative_type=fire provided that the target is more sensitive to fire than to the original type of the attack

Reply:
@Pentarctagon I think this line is essential for the proper functioning of the code, as for the other attributes of the attack

Reply:
Line removed because unnecessary.